flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java

```diff
@@ -121,8 +122,11 @@
 	public static final ConfigOption<String> CONTAINER_IMAGE =
 		key("kubernetes.container.image")
 		.stringType()
-		.defaultValue("flink:latest")
-		.withDescription("Image to use for Flink containers.");
+		.defaultValue("flink:" + Version.PROJECT_VERSION + "-scala_" + Version.SCALA_VERSION)
```

Comment:
Could we simply use `EnvironmentInformation#getVersion()` to get the project version? And i think we do not need the scalar version. Since we always have a image tag corresponding to the Flink version(with scala 2.12 built in).

Reply:
I reported https://issues.apache.org/jira/browse/FLINK-16289 last week about getting java.io.InvalidClassException: org.apache.flink.table.codegen.GeneratedAggregationsFunction.
Root cause after several days of digging:
- I had scala 2.11 on my machine
- Downloaded Flink 1.10.0 for scala 2.11 to start my session on Kubernetes.
- Used Flink 1.10.0 for scala 2.11 for all dependencies in my project that I built and ran.
- Then "Flink 1.10.0 for scala 2.11" on Kubernetes downloaded "flink:latest" image which is really "Flink 1.10.0 for scala 2.12" which broke it all.

Also if for some reason I choose not to upgrade and a new release for Flink is created (i.e. 1.11) then my application may also break.

So I think "Flink X for scala Y" must download the docker image for exactly THAT combination of versions.

So I think having the scala version in there explicitly is a must have.

or ... We ditch the scala 2.11 variant completely and have only 1 scala variant.

I'll have a look at the EnvironmentInformation you mentioned, perhaps that is the better place to add the scala version.